docs/aws/audit/iamcompliance/rules/root_account_access_keys.mdx

```diff
@@ -23,6 +23,214 @@ HIPAA, ISO27001, PCIDSS, GDPR, NIST, SOC2, CISAWS, CBP, HITRUST, AWSWAF, NISTCSF
+
+1. **Check for Existing Access Keys:**
+   First, you need to check if there are any access keys associated with the root account. This can be done by listing the access keys for the root user.
+   ```sh
+   aws iam list-access-keys --user-name root
+   ```
+
+2. **Delete Existing Access Keys:**
+   If there are any access keys associated with the root account, you should delete them. Replace `ACCESS_KEY_ID` with the actual access key ID you want to delete.
+   ```sh
+   aws iam delete-access-key --user-name root --access-key-id ACCESS_KEY_ID
+   ```
+
+3. **Create an IAM User for Administrative Tasks:**
+   Instead of using the root account, create an IAM user with administrative privileges. This user will be used for tasks that require elevated permissions.
+   ```sh
+   aws iam create-user --user-name AdminUser
+   aws iam attach-user-policy --user-name AdminUser --policy-arn arn:aws:iam::aws:policy/AdministratorAccess
+   ```
+
+4. **Enable MFA for Root Account:**
+   To further secure the root account, enable Multi-Factor Authentication (MFA). This step doesn't directly involve the CLI but is a best practice to ensure the root account is secure.
+   ```sh
+   aws iam enable-mfa-device --user-name root --serial-number arn:aws:iam::123456789012:mfa/root-account-mfa-device --authentication-code1 123456 --authentication-code2 789012
+   ```
+
+By following these steps, you can ensure that the root account does not have access keys, and you can use an IAM user with administrative privileges for necessary tasks.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the root account from having access keys in IAM using Python scripts, you can use the AWS SDK for Python (Boto3). Here are the steps to achieve this:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. You can install it using pip if you haven't already:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session with the necessary credentials and region information.
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='YOUR_REGION'
+   )
+   ```
+
+3. **Check for Root Access Keys**:
+   Use the IAM client to list access keys for the root account and ensure no access keys exist.
+   ```python
+   iam_client = session.client('iam')
+
+   # List access keys for the root account
+   response = iam_client.list_access_keys(UserName='root')
+
+   # Check if any access keys exist
+   if response['AccessKeyMetadata']:
+       print("Root account has access keys. Please remove them.")
+   else:
+       print("Root account does not have any access keys.")
+   ```
+
+4. **Automate the Check and Preventive Action**:
+   You can automate this check to run periodically and alert or take action if access keys are found.
+   ```python
```

Comment:
<h2>Secret Detected: Secret Keyword</h2> <a href="https://console.cloudanix.com/code?t=secrets"> Click here for more details</a>